integrations/argocd/main.py

```diff
@@ -48,7 +48,7 @@ async def on_managed_resources_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
             application_name=application["metadata"]["name"]
         )
         application_resource = [
-            {**managed_resource, "__applicationId": application["metadata"]["uid"]}
+            {**managed_resource, "__application": application}
```

Comment:
this breaks backwards compatibility with users that were using `__applicationId`. Lets add extra `__application` and still leave `__applicationId`. we will need to somehow indicate to the user maybe in the docs, that `__applicationId` will be removed in the future. so also add a TODO comment above the `__applicationId`